_posts/2015-06-25-infrastructure-as-code.md

```diff
@@ -0,0 +1,46 @@
+
+# Jenkins configuration
+
+Jenkins makes it very easy to introduce new functionality by installing one of the [many available plugins](https://wiki.jenkins-ci.org/display/JENKINS/Plugins). Unfortunately not every plugin turns out to work as expected or in the worst case can potentially bring down the whole Jenkins instance. It is a bad idea to evaluate unknown plugins by installing them into your production CI infrastructure. Even making changes to a job configuration without adding any new plugins can lead to broken build jobs, negatively affecting the productivity of whole teams.
+
+## The solution
+
+We were looking for ways to make it safe and easy to improve our build infrastructure. Using version control like [GitHub](https://github.com/) for all of our source code it was natural to treat our infrastructure the same way and be able to roll back to a previous version without any hassle. Versioning whole images of Jenkins virtual servers didn't sound like the best idea and plugins like [JobConfigHistory](https://wiki.jenkins-ci.org/display/JENKINS/JobConfigHistory+Plugin) didn't go far enough.
+
+
+## Server provisioning using Ansible and Vagrant
+
+We started by tackling the perceived root problem: the installation of the Jenkins master, including the required plugins.  The idea is to be able to set up a Jenkins master from scratch by just running a single script, while still being able to use the same facilities to update the installation, e.g. with new plugins, using the same technology. To achieve this, we decided to use [Ansible](http://www.ansible.com/home), mainly due to existing know-how.
+
+Once the basic installation of Jenkins master and it's prerequisites is done, the interesting part begins: installing & configuring plugins. The installation of the additional plugins is done by utilizing the [Jenkins-CLI](https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+CLI). This allows you to provide plugin names and versions to be installed. The tricky part is getting the configuring of the server right. For this, we have so far not found a more clever solution than just to create it locally (but templated) and copy it over to the master using the according Ansible task. The addition of nodes can be done via Jenkins-CLI again.
```

Comment:
Replace the "&" by "and": installing and configuring plugins.

Use British English: utilising